nomt/src/cursor.rs

```diff
@@ -298,15 +298,8 @@ struct PagePrefetcher {
 
 impl PagePrefetcher {
     fn new(dest: KeyPath) -> Self {
-        let rev_page_ids = {
-            // Chop the destination key path into the corresponding page IDs and reverse them so
-            // that we can `pop`.
-            let mut x = PageIdsIterator::new(dest).collect::<Vec<_>>();
-            x.reverse();
-            x
-        };
         Self {
-            rev_page_ids,
+            page_ids_iter: PageIdsIterator::new(dest),
```

Comment:
Could we please `reserve(DEPTH)` while we are at it? 🥺👉👈

Reply:
do you mean `page_promises: Vec::with_capacity(PREFETCH_N)`?

Reply:
Yup!